CI is broken because a few dependencies are not downloading.

**Describe the bug**
CI is broken because a few dependencies are not downloading. See  https://github.com/kiwix/kiwix-android/actions/runs/20744634527/job/59564444875?pr=4577

**Logs**
```kotlin
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'kiwix-android'.
> Could not resolve all artifacts for configuration 'classpath'.
   > Could not resolve com.squareup.okio:okio:2.1.0.
     Required by:
         root project : > de.fayard.buildSrcVersions:de.fayard.buildSrcVersions.gradle.plugin:0.7.0 > de.fayard:plugin:0.7.0
      > Could not resolve com.squareup.okio:okio:2.1.0.
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/squareup/okio/okio/2.1.0/okio-2.1.0.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/squareup/okio/okio/2.1.0/okio-2.1.0.pom'. Received status code 403 from server: Forbidden
      > Could not resolve com.squareup.okio:okio:2.1.0.
         > Could not get resource 'https://plugins.gradle.org/m2/com/squareup/okio/okio/2.1.0/okio-2.1.0.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/squareup/okio/okio/2.1.0/okio-2.1.0.pom'. Received status code 403 from server: Forbidden
   > Could not resolve com.squareup.moshi:moshi:1.7.0.
     Required by:
         root project : > de.fayard.buildSrcVersions:de.fayard.buildSrcVersions.gradle.plugin:0.7.0 > de.fayard:plugin:0.7.0
      > Could not resolve com.squareup.moshi:moshi:1.7.0.
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/squareup/moshi/moshi/1.7.0/moshi-1.7.0.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/squareup/moshi/moshi/1.7.0/moshi-1.7.0.pom'. Received status code 403 from server: Forbidden
      > Could not resolve com.squareup.moshi:moshi:1.7.0.
         > Could not get resource 'https://plugins.gradle.org/m2/com/squareup/moshi/moshi/1.7.0/moshi-1.7.0.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/squareup/moshi/moshi/1.7.0/moshi-1.7.0.pom'. Received status code 403 from server: Forbidden
   > Could not resolve com.squareup:kotlinpoet:1.3.0.
     Required by:
         root project : > de.fayard.buildSrcVersions:de.fayard.buildSrcVersions.gradle.plugin:0.7.0 > de.fayard:plugin:0.7.0
      > Could not resolve com.squareup:kotlinpoet:1.3.0.
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/squareup/kotlinpoet/1.3.0/kotlinpoet-1.3.0.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/squareup/kotlinpoet/1.3.0/kotlinpoet-1.3.0.pom'. Received status code 403 from server: Forbidden
      > Could not resolve com.squareup:kotlinpoet:1.3.0.
         > Could not get resource 'https://plugins.gradle.org/m2/com/squareup/kotlinpoet/1.3.0/kotlinpoet-1.3.0.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/squareup/kotlinpoet/1.3.0/kotlinpoet-1.3.0.pom'. Received status code 403 from server: Forbidden
   > Could not resolve commons-codec:commons-codec:1.11.
     Required by:
         root project : > com.android.tools.build:gradle:8.11.1 > com.android.tools.analytics-library:crash:31.11.1 > org.apache.httpcomponents:httpclient:4.5.14
      > Could not resolve commons-codec:commons-codec:1.11.
         > Could not get resource 'https://repo.maven.apache.org/maven2/commons-codec/commons-codec/1.11/commons-codec-1.11.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/commons-codec/commons-codec/1.11/commons-codec-1.11.pom'. Received status code 403 from server: Forbidden
      > Could not resolve commons-codec:commons-codec:1.11.
         > Could not get resource 'https://plugins.gradle.org/m2/commons-codec/commons-codec/1.11/commons-codec-1.11.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/commons-codec/commons-codec/1.11/commons-codec-1.11.pom'. Received status code 403 from server: Forbidden
3 actionable tasks: 3 executed
   > Could not resolve com.thoughtworks.xstream:xstream:1.4.10.
     Required by:
         root project : > de.fayard.buildSrcVersions:de.fayard.buildSrcVersions.gradle.plugin:0.7.0 > de.fayard:plugin:0.7.0 > com.github.ben-manes:gradle-versions-plugin:0.25.0
      > Could not resolve com.thoughtworks.xstream:xstream:1.4.10.
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/thoughtworks/xstream/xstream/1.4.10/xstream-1.4.10.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/thoughtworks/xstream/xstream/1.4.10/xstream-1.4.10.pom'. Received status code 403 from server: Forbidden
      > Could not resolve com.thoughtworks.xstream:xstream:1.4.10.
         > Could not get resource 'https://plugins.gradle.org/m2/com/thoughtworks/xstream/xstream/1.4.10/xstream-1.4.10.pom'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/thoughtworks/xstream/xstream/1.4.10/xstream-1.4.10.pom'. Received status code 403 from server: Forbidden

```


Some more dependency issues. It seems the issue is something else. However, downloading directly by opening these URL works.

```kotlin
FAILURE: Build failed with an exception.
* What went wrong:
A problem occurred configuring root project 'kiwix-android'.
> Could not resolve all artifacts for configuration 'classpath'.
   > Could not download compose-compiler-gradle-plugin-2.0.0-gradle85.jar (org.jetbrains.kotlin:compose-compiler-gradle-plugin:2.0.0)
      > Could not get resource 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/compose-compiler-gradle-plugin/2.0.0/compose-compiler-gradle-plugin-2.0.0-gradle85.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/compose-compiler-gradle-plugin/2.0.0/compose-compiler-gradle-plugin-2.0.0-gradle85.jar'. Received status code 403 from server: Forbidden
   > Could not download kotlin-gradle-plugin-api-2.0.0-gradle85.jar (org.jetbrains.kotlin:kotlin-gradle-plugin-api:2.0.0)
      > Could not get resource 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin-api/2.0.0/kotlin-gradle-plugin-api-2.0.0-gradle85.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-gradle-plugin-api/2.0.0/kotlin-gradle-plugin-api-2.0.0-gradle85.jar'. Received status code 403 from server: Forbidden
   > Could not download keeper-0.16.1.jar (com.slack.keeper:keeper:0.16.1)
      > Could not get resource 'https://repo.maven.apache.org/maven2/com/slack/keeper/keeper/0.16.1/keeper-0.16.1.jar'.
3 actionable tasks: 3 executed
         > Could not GET 'https://repo.maven.apache.org/maven2/com/slack/keeper/keeper/0.16.1/keeper-0.16.1.jar'. Received status code 403 from server: Forbidden
   > Could not download commons-codec-1.11.jar (commons-codec:commons-codec:1.11)
      > Could not get resource 'https://repo.maven.apache.org/maven2/commons-codec/commons-codec/1.11/commons-codec-1.11.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/commons-codec/commons-codec/1.11/commons-codec-1.11.jar'. Received status code 403 from server: Forbidden
   > Could not download xstream-1.4.10.jar (com.thoughtworks.xstream:xstream:1.4.10)
      > Could not get resource 'https://repo.maven.apache.org/maven2/com/thoughtworks/xstream/xstream/1.4.10/xstream-1.4.10.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/com/thoughtworks/xstream/xstream/1.4.10/xstream-1.4.10.jar'. Received status code 403 from server: Forbidden
   > Could not download xmlpull-1.1.3.1.jar (xmlpull:xmlpull:1.1.3.1)
      > Could not get resource 'https://repo.maven.apache.org/maven2/xmlpull/xmlpull/1.1.3.1/xmlpull-1.1.3.1.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/xmlpull/xmlpull/1.1.3.1/xmlpull-1.1.3.1.jar'. Received status code 403 from server: Forbidden
   > Could not download xpp3_min-1.1.4c.jar (xpp3:xpp3_min:1.1.4c)
      > Could not get resource 'https://plugins.gradle.org/m2/xpp3/xpp3_min/1.1.4c/xpp3_min-1.1.4c.jar'.
         > Could not GET 'https://repo.maven.apache.org/maven2/xpp3/xpp3_min/1.1.4c/xpp3_min-1.1.4c.jar'. Received status code 403 from server: Forbidden
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -25,18 +25,18 @@ jobs:
           swap-storage: false
 
       - name: checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           distribution: temurin
           java-version: 17
 
       - name: Restore Cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         with:
           path: |
             ~/.gradle/caches
@@ -52,7 +52,7 @@ jobs:
           sudo udevadm trigger --name-match=kvm
 
       - name: AVD cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         id: avd-cache
         with:
           path: |
@@ -98,7 +98,7 @@ jobs:
           script: bash contrib/instrumentation.sh
 
       - name: Upload screenshot result
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v6
         if: failure()
         with:
           name: ${{ matrix.api-level }}
@@ -110,12 +110,12 @@ jobs:
 
       - name: Upload coverage to Codecov
         if: ${{ matrix.api-level==30 }}
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v5
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
 
       - name: Upload Coverage to GH-Actions
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v6
         if: ${{ matrix.api-level==30 }}
         with:
           name: Tests Coverage Report
@@ -139,18 +139,18 @@ jobs:
           swap-storage: false
 
       - name: checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           distribution: temurin
           java-version: 17
 
       - name: Restore Cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         with:
           path: |
             ~/.gradle/caches
@@ -166,7 +166,7 @@ jobs:
           sudo udevadm trigger --name-match=kvm
 
       - name: AVD cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         id: avd-cache
         with:
           path: |
@@ -228,18 +228,18 @@ jobs:
           swap-storage: false
 
       - name: checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           distribution: temurin
           java-version: 17
 
       - name: Restore Cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         with:
           path: |
             ~/.gradle/caches
@@ -255,7 +255,7 @@ jobs:
           sudo udevadm trigger --name-match=kvm
 
       - name: AVD cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         id: avd-cache
         with:
           path: |
@@ -317,18 +317,18 @@ jobs:
           swap-storage: false
 
       - name: checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           distribution: temurin
           java-version: 17
 
       - name: Restore Cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         with:
           path: |
             ~/.gradle/caches
@@ -344,7 +344,7 @@ jobs:
           sudo udevadm trigger --name-match=kvm
 
       - name: AVD cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         id: avd-cache
         with:
           path: |

--- a/.github/workflows/dummy_bundle_and_apk.yml
+++ b/.github/workflows/dummy_bundle_and_apk.yml
@@ -11,10 +11,10 @@ jobs:
     runs-on: ubuntu-22.04
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin
@@ -69,13 +69,13 @@ jobs:
           echo "apk_dir=$APK_DIR" >> $GITHUB_ENV
 
       - name: Upload Bundle as an artifact
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v6
         with:
           name: ${{ env.bundle_name }}
           path: ${{ env.bundle_path }}
 
       - name: Upload All Release APKs as artifacts
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v6
         with:
           name: ReleaseApks
           path: ${{ env.apk_dir }}*.apk

--- a/.github/workflows/fdroid_nightly.yml
+++ b/.github/workflows/fdroid_nightly.yml
@@ -12,10 +12,10 @@ jobs:
     environment: nightly
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v6
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin

--- a/.github/workflows/nightly.yml
+++ b/.github/workflows/nightly.yml
@@ -12,12 +12,12 @@ jobs:
     steps:
 
       - name: checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin

--- a/.github/workflows/pull_request.yml
+++ b/.github/workflows/pull_request.yml
@@ -13,18 +13,18 @@ jobs:
     continue-on-error: true
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin
 
       - name: Restore Cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         with:
           path: |
             ~/.gradle/caches
@@ -38,7 +38,7 @@ jobs:
 
       - name: Upload ktlint Reports
         if: ${{ always() }}
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v6
         with:
          name: ktlint-report
          path: '**/build/reports/ktlint'
@@ -51,18 +51,18 @@ jobs:
     continue-on-error: true
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin
 
       - name: Restore Cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         with:
           path: |
             ~/.gradle/caches
@@ -74,7 +74,7 @@ jobs:
       - name: Static Analysis
         run: ./gradlew detektDebug detektCustomExampleDebug
       - name: Upload Lint Reports
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v6
         if: ${{ always() }}
         with:
           name: detekt-report
@@ -89,18 +89,18 @@ jobs:
     steps:
 
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin
 
       - name: Restore Cache
-        uses: actions/cache@v4
+        uses: actions/cache@v5
         with:
           path: |
             ~/.gradle/caches
@@ -112,7 +112,7 @@ jobs:
         run: ./gradlew app:lintDebug custom:lintCustomexampleDebug
 
       - name: Upload Lint Reports
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v6
         if: ${{ always() }}
         with:
           name: lint-report
@@ -127,12 +127,12 @@ jobs:
     continue-on-error: true
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
         with:
           fetch-depth: 1
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin
@@ -149,7 +149,7 @@ jobs:
         run: ./gradlew assemble
 
       - name: Upload APK as Artifacts
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v6
         if: ${{ always() }}
         with:
             name: kiwix-android

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -10,10 +10,10 @@ jobs:
     runs-on: ubuntu-22.04
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin

--- a/.github/workflows/testing_release.yml
+++ b/.github/workflows/testing_release.yml
@@ -14,10 +14,10 @@ jobs:
     runs-on: ubuntu-22.04
 
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v6
 
       - name: Set up JDK 17
-        uses: actions/setup-java@v4
+        uses: actions/setup-java@v5
         with:
           java-version: 17
           distribution: temurin

